includes/class-wc-download-handler.php

```diff
@@ -41,8 +41,26 @@ public static function download_product() {
 			self::download_error( __( 'Invalid download link.', 'woocommerce' ) );
 		}
 
+		// Fallback, accept email address if it's passed.
+		if ( empty( $_GET['email'] ) && empty( $_GET['uid'] ) ) {
+			self::download_error( __( 'Invalid download link.', 'woocommerce' ) );
+		}
+
+		if ( isset( $_GET['email'] ) ) {
+			$email_address = $_GET['email'];
+		} else {
+			// Get email address from order to verify hash.
+			$order_id = wc_get_order_id_by_order_key( $_GET['order'] );
+			$order = new WC_Order( $order_id );
+			$email_address = $order->get_billing_email();
+
+			if( $_GET['uid'] !== hash( 'sha256', $email_address ) ) {
```

Comment:
Missing space after `if`, and should change `!==` to `hash_equals()`.

Reply:
Good catch. 👍